tensorboard/components/tf_color_scale/colorScale.ts

```diff
@@ -63,14 +63,24 @@ export class ColorScale {
  * A color scale whose domain is the set of runs currently available.
  * Automatically updated as this data changes.
  */
-const _runsColorScale = new ColorScale();
-export const runsColorScale: ((runName: string) => string) =
-    _runsColorScale.scale.bind(_runsColorScale);
-
-function updateRunsColorScale(): void {
-  _runsColorScale.domain(tf_backend.runsStore.getRuns());
+function createAutoUpdateColorScale(
+    store: tf_backend.BaseStore,
+    getDomain: () => string[]): (runName: string) => string {
+  const colorScale = new ColorScale();
+  function updateRunsColorScale(): void {
+    colorScale.domain(getDomain());
+  }
+  store.addListener(updateRunsColorScale);
+  updateRunsColorScale();
+  return (runName) => colorScale.scale(runName);
```

Comment:
Optional but since you're calling the method `getColor()` in the places where we're actually using the color scale now, maybe rename `scale()` here to `getColor()` as well for consistency?  The name `scale()` was never a great name anyway since it's ambiguous whether it's intended as a verb or a noun.